src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/NewAzureStorageAccount.cs

```diff
@@ -134,6 +134,23 @@ public PSNetworkRuleSet NetworkRuleSet
             get; set;
         }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Enable HierarchicalNamespace for the Storage account.")]
+        [ValidateNotNullOrEmpty]
+        public bool EnableHierarchicalNamespace
```

Comment:
Use [`SwitchParameter`](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.switchparameter?view=powershellsdk-1.1.0) instead of Boolean... that way you don't have to do the null check.

Reply:
Actually, we really have a long discusstion with Mark for this, and finally decided to use bool. 
In a word, since we need 3 status: true/false/notSet, we need use bool.

Reply:
@blueww thanks for the clarification.